src/plugins/data/server/ui_settings.ts

```diff
@@ -762,5 +762,19 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
       }),
       schema: schema.arrayOf(schema.string()),
     },
+    [UI_SETTINGS.SEARCH_USE_FIELDS]: {
+      name: i18n.translate('data.advancedSettings.searchUseFieldsTitle', {
+        defaultMessage: 'Include all fields in search request',
+      }),
+      value: false,
+      description: i18n.translate('data.advancedSettings.searchUseFieldsDescription', {
+        defaultMessage: `
+        <strong>Experimental</strong>:
+        Adds the <code>"fields": ["*"]</code> property to search request body`,
+      }),
+      schema: schema.boolean(),
+      category: ['search'],
+      requiresPageReload: true,
```

Comment:
why does this require reload?

Reply:
Just checked with @kavilla, `search_source` should access the dependencies every time it calls `flatten()`, so we don't need it. Nice catch.